vmm/src/interrupt.rs

```diff
@@ -272,30 +272,37 @@ impl InterruptSourceGroup for LegacyUserspaceInterruptGroup {
     }
 }
 
-pub struct KvmInterruptManager {
+pub struct KvmLegacyUserspaceInterruptManager {
+    ioapic: Option<Arc<Mutex<ioapic::Ioapic>>>,
```

Comment:
Don't use `Option<>` since you simplified the whole thing by having dedicated interrupt managers.
The option was only meant to be used when the same structure was supporting both with and without Ioapic.

Reply:
Ah right, good catch. More simplification, let me fix that.

Reply:
Done.